distributed/deploy/spec.py

```diff
@@ -77,6 +77,19 @@ def __repr__(self):
         await self.close()
 
 
+class NoOpAwaitable(object):
+    """An awaitable object that always returns None.
+
+    Useful to return from a method that can be called in both asynchronous and
+    synchronous contexts"""
+
+    def __await__(self):
+        async def f():
+            return None
+
+        return f().__await__()
```

Comment:
We might also consider `asyncio.sleep(0)` or `tornado.gen.moment`

Although this might be nicer if either of the two return something, which would look bad in Jupyter.  I suspect that they're clean though.

Reply:
> We might also consider asyncio.sleep(0) or tornado.gen.moment

Why? In this case there's no context switch. Asyncio's event loop uses an eager scheduler - if an async function is called that doesn't await on anything that would block (e.g. `sleep`, I/O, etc...) then things run like normal function calls and return immediately. Inserting a `sleep(0)` or `gen.moment()` (which is deprecated, BTW) would allow another coroutine to resume execution, but I don't see that as necessary here.

We also only return a `NoOpAwaitable` if the cluster is running in async mode, so in Jupyter notebook contexts there'd be no output.